content/engineering/guest-post/how-to-implement-jwt-authentication-in-deno/index.md

```diff
@@ -0,0 +1,656 @@
+
+`middlewares.isAuthorized.ts`
+
+```js
+import { verify } from "https://deno.land/x/djwt@v2.4/mod.ts";
+import { key } from "../utils/apiKey.ts";
+import { Context } from "https://deno.land/x/oak/mod.ts";
+
+export const authourized = async (ctx: Context, next:any) => {
+    try{   
+    const headers: Headers = ctx.request.headers;
+    const authorization = headers.get('Authorization');
+    if(!authorization) {
+        ctx.response.status = 401;
+        return;
+    }
+    const jwt = authorization.split(' ')[1];
+
+    if(!jwt) {
+        ctx.response.status = 401;
+        return;
+    }
+    const payload = await verify(jwt, key);
+
+    if(!payload){
+    throw new Error("!payload")
+    }
+     await next();
+     
+     } catch (error) {
+        ctx.response.status = 401;
+        ctx.response.body ={message: "You are not authorized to access this route"}
+        return;
+    }
+};
+```
+
+To protect your todo routes, import the authorized middleware function from the `isAuthorized.ts` middleware file and pass the middleware into the todo routes to protect them from unauthorized people.
+
+`routes.allRoutes.ts`
+
+```js
+import { Router } from "https://deno.land/x/oak/mod.ts";
+import {signup, signin} from "../controllers/users.ts";
+import {create, getTasks, getById, updateById, deleteTask} from "../controllers/tasks.ts";
+import { authourized } from "../middlewares/isAuthorized.ts";
+
+const router = new Router();
+
+//User routes
+router.post("/api/signup", signup)
+      .post("/api/signin", signin);
+
+//Task routes
+router.post("/api/tasks", authourized, create)
+      .get("/api/tasks", authourized, getTasks)
+      .get("/api/tasks/:taskId", authourized, getById)
+      .patch("/api/tasks/:taskId", authourized, updateById)
+      .delete("/api/tasks/:taskId", authourized, deleteTask);
+
+export default router;
+```
+
+Awesome! Your application is ready, but wait! Don't get too excited yet, relax let's test the application :)
+
+Below is what your final project directory looks like:
+
+`DenoAPI_JWT_Auth`
+│
+└─**src**
+│   └───controllers
+│   │      └───users.ts
+│   │      └───tasks.ts
+│   │
+│   └───database
+│   │      └───connectDB.ts
+│   │         
+│   └───middlewares
+│   │      └───isAuthorized.ts
+│   │         
+│   └───routes
+│   │      └───allRoutes.ts
+│   │
+│   └───schema
+│   │      └───user.ts
+│   │      └───task.ts
+│   │ 
+│   └───utils
+│   │      └───apiKey.ts
+│   │         
+└─**app.ts**
+
+## Test Application
+
+Now that your application is ready, you need to test the various routes to ensure they are working. To test the routes, run your server again with `deno run --allow-net app.ts`.
+
+![deno run screenshot](./aPk23wT.png)
+
+Great! The app is running on port:8080. You can now head to the postman to test your routes.
+
+### Signup Route
+
+![Sign up](./D2ORbZ0.png)
+
+![](./trYDtXn.png)
```

Comment:
@icode247 Seems like the image specified here is missing in the folder. Please cross-check.

Reply:
I have added the missing image